### PR TITLE
Add api to fetch competencies

### LINKED ISSUE #272: An API endpoint is needed for fetching pages of competencies
## Expected behaviour

An HTTP `GET` request to `/competencies` responds with a paginated list of competencies from the database.

## Actual behaviour

There is no way to fetch competencies from the database via the API.

## Details and resources

There are helpers on the backend (middleware) for handling paginated collections.

There is a pattern of having database queries in the services layers, and validation in the router layer.

## Checklist

- [x] This issue is linked to the appropriate project.
- [x] This issue has appropriate labels.

### PR DESCRIPTION
**Pull request by BRashad**
## Proposed changes

- Add `countCompetencies` function into competencies service
- Add test for `countCompetencies` function in competencies service test
- Add `competenciesRouter`  get a request to show a paginated list of competencies
- Add `competenciesRouter`  get request test to show a paginated list of competencies and check/throw errors when counting or listing competencies.
- Add `competencies` routes in server file 

Resolves #272 

## Checklist

- [X] Are the issues being addressed linked to this PR?
- [X] Do all commit messages start with the issue number?
- [X] Are all code changes sufficiently tested?
- [X] Are there screenshots for UI changes? (N/A)

### CODE REVIEW

File: api/src/middleware/competenciesRouter.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import { Router } from 'express';
```
Comment (pthan1) [inline api/src/middleware/competenciesRouter.ts:1]: I think we should follow the previous convention and add a line between lines 1 and 2.

File: api/src/middleware/__tests__/competenciesRouter.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+          err => {
+            expect(mockCountCompetencies).toHaveBeenCalled();
+            expect(mockListCompetencies).toHaveBeenCalledWith(1, 1);
+            done(err);
+          }
+        );
+    });
+
+    it('should respond with an internal server error if an error was thrown while listing competencies', done => {
+      mockListCompetencies.mockRejectedValue(new Error());
```
Comment (pthan1) [inline api/src/middleware/__tests__/competenciesRouter.ts:43]: This follows the format in the meetingsRouter test.  However, the test for responding with an error if an error was thrown while listing meetings has mock data for the countMeetings function.  

Would that sort of code code be valuable to include in this test?

**Review by pthan1 — APPROVED** [review]
Awesome job figuring out why your countCompetencies test wasn't passing due to the bindings that were passed through the mockQuery function.  The services/router look solid and follow convention.

The Router test does not include a test if an error was thrown while counting meetings. Since that function was needed for the issue, should that be tested in this PR as well?

File: api/src/middleware/competenciesRouter.ts (hunk 1 above)
Comment (BRashad) [reply api/src/middleware/competenciesRouter.ts:1]: ok we fixed it.

File: api/src/middleware/__tests__/competenciesRouter.ts (hunk 2 above)
Comment (eileenxue) [reply api/src/middleware/__tests__/competenciesRouter.ts:43]: Hi Phil, thanks for picking that up. We recently added the countCompetencies function, and forgot to include some tests for counting competencies errors. We'll add that in.

**Review by LouisDG6 — APPROVED** [review]
Looks good and nice work following convention!

**Comment by BRashad** [comment]
@davidvandusen we created a new route for the competencies to make HTTP requests from.

File: api/src/middleware/__tests__/competenciesRouter.ts [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+import competenciesRouter from '../competenciesRouter';
+
+jest.mock('../../services/competenciesService.ts');
+const mockCountCompetencies = jest.mocked(countCompetencies);
+const mockListCompetencies = jest.mocked(listCompetencies);
+
+describe('competenciesRouter', () => {
+  const appAgent = createAppAgentForRouter(competenciesRouter);
+
+  describe('GET /', () => {
```
Comment (pthan1) [inline api/src/middleware/__tests__/competenciesRouter.ts:16]: @BRashad The route in the Describe block is incorrect.  I believe it should say '/competencies', no?

**Comment by BRashad** [comment]
@davidvandusen just changed the alphabetical order of routes

**Comment by BRashad** [comment]
@davidvandusen Yes. we tested the endpoint with real data, it works. please see the screenshot.

Screen Shot 2022-02-02 at 1 11 57 PM

**Pull request merged**